guides/source/association_basics.md

```diff
@@ -478,16 +619,37 @@ class Part < ApplicationRecord
 end
 ```
 
-The simplest rule of thumb is that you should set up a `has_many :through` relationship if you need to work with the relationship model as an independent entity. If you don't need to do anything with the relationship model, it may be simpler to set up a `has_and_belongs_to_many` relationship (though you'll need to remember to create the joining table in the database).
+You'd use `has_many :through` when:
 
-You should use `has_many :through` if you need validations, callbacks, or extra attributes on the join model.
+- You need to add extra attributes or methods to the join table.
+- You require validations or callbacks on the join model.
+- The join table should be treated as an independent entity with its own behavior.
 
-While `has_and_belongs_to_many` suggests creating a join table with no primary key via `id: false`, consider using a composite primary key for the join table in the `has_many :through` relationship.
-For example, it's recommended to use `create_table :manifests, primary_key: [:assembly_id, :part_id]` in the example above.
+The `has_and_belongs_to_many` association allows you to create a many-to-many relationship directly between two models without needing an intermediary model. This method is straightforward and is suitable for simple associations where no additional attributes or behaviors are required on the join table. For `has_and_belongs_to_many` associations, you'll need to create a join table without a primary key.
+
+```ruby
+class Assembly < ApplicationRecord
+  has_and_belongs_to_many :parts
+end
+
+class Part < ApplicationRecord
+  has_and_belongs_to_many :assemblies
+end
+```
+
+You'd use `has_and_belongs_to_many` when:
+
+- The association is simple and does not require additional attributes or behaviors on the join table.
+- You do not need validations, callbacks, or extra methods on the join table.
+
+Other Associations
+-------------------------
```

Comment:
Maybe some of these sub sections can be upgraded to top sections instead?
It seems "Polymorphic Associations", "Self-Joins", "STI" and "Delegated Types" are somewhat on the same abstraction level?

Reply:
> Maybe some of these sub-sections can be upgraded to top sections instead?
It seems "Polymorphic Associations", "Self-Joins", "STI" and "Delegated Types" are somewhat on the same abstraction level?

I would have liked to group them but the sub headings for STI and Delegated types are useful, and we lose the index under those headings once we move them under another subheading like Advanced Associations. 

Maybe another option would be to bring them closer up to the top, after "Advanced Associations", so that they'll be sections 5 and 6 and everything else will move further down. 

What are your thoughts?

Reply:
Moving them after "Advanced Assciations" sounds great!